src/components/Articles.jsx

```diff
@@ -3,30 +3,59 @@ import { ArticleContext } from "../contexts/Article";
 import { getArticles } from "../api";
 import ArticleCard from "./ArticleCard";
 import { useParams } from "react-router-dom";
+import { Box, FormControl, InputLabel, MenuItem, Select } from "@mui/material";
 
 const Articles = () => {
   const { articles, setArticles } = useContext(ArticleContext);
   const [isLoading, setIsLoading] = useState(true);
+  const [sortBy, setSortBy] = useState("votes");
   const { slug } = useParams();
 
   useEffect(() => {
     setIsLoading(true);
     getArticles(slug).then((articlesData) => {
-      setArticles(articlesData.articles);
+      const sortArticles = [...articlesData.articles].sort(
+        (a, b) => b[sortBy] - a[sortBy]
```

Comment:
We shouldn't be doing this in the front end, you have build an endpoint in the backend that makes use of queries, so please use it! 😄